### ISSUE TITLE
dart2js in latest dev release breaks Observatory (and therefore Fuchsia)

### ISSUE THREAD

**Issue by rmacnak-google** [dart-lang/sdk#32099]
(Currently this is 2.0.0-dev.22.0)

```
./tools/download_latest_dev_sdk.py
rm -rf out/ReleaseX64/
./tools/build.py -mrelease runtime

...
Building observatory...
[Info from Dart2JS]:
Compiling observatory|web/main.dart...
[Error from Dart2JS on observatory|web/main.dart]:
web/packages/observatory/src/elements/function_view.dart:61:5:
Expected an identifier, but got 'Function'.
  M.Function _function;
    ^^^^^^^^
[Error from Dart2JS on observatory|web/main.dart]:
web/packages/observatory/src/elements/isolate_view.dart:68:5:
Expected an identifier, but got 'Function'.
  M.Function _function;
    ^^^^^^^^
```

**Comment by rmacnak-google** [comment]
@whesse @sigmundch

**Comment by sigmundch** [comment]
@rmacnak-google I'm curious, why is this not showing up in the bots? Doesn't observatory get built on our bots on every commit?

**Comment by rmacnak-google** [comment]
The version of the pre-build SDK used by dart-lang/sdk is pinned by tools/sdks/$os/dart-sdk.tar.gz.sha1. Fuchsia was using `latest` as the version. We've now pinned Fuchsia to `2.0.0-dev.21.0`.

**Comment by rmacnak-google** [comment]
I'm confused why the suite "observatory_ui" still seems to be passing on the chrome bots though.

**Comment by whesse** [comment]
You can see which configurations are running the unit tests by checking the
test matrix.  I'll try and get a list to you. Maybe the unit tests are
still working. What is broken, exactly?  The solution will be to fix it and
release tomorrow.

**Comment by sigmundch** [comment]
I bisected the issue and it appears this was caused by this recent FE change: b2c5415db90f1262d0cbec918b3d57543045bb5a.

@danrubel - could you take a look?

Here is how to reproduce the error:
```
  > cd runtime/observatory/
  > pub get
  > cd ../../
  > ./sdk/bin/dart2js runtime/observatory/web/main.dart
```

You'll notice that the compile step is successful on the commit above b2c5415db90f1262d0cbec918b3d57543045bb5a.

**Comment by danrubel** [comment]
Working on a fix for this... https://dart-review.googlesource.com/c/sdk/+/40320

**Comment by danrubel** [comment]
This has landed. Please confirm that the issue is fixed.

**Comment by sigmundch** [comment]
confirmed! :)

**Comment by whesse** [comment]
Fix is in SDK version 2.0.0-dev.23.0.
That roll includes all commits up to a0795f185052bc927f5564470dd156a0a2d4bf1c at 19:30 UTC Feb 9 2017, including 87a395bd0986cee3261e780e9522bc9fc8ee872c.
